pennylane/ops/op_math/exp.py

```diff
@@ -280,3 +288,162 @@ def simplify(self):
         if isinstance(self.base, qml.ops.op_math.SProd):  # pylint: disable=no-member
             return Exp(self.base.base.simplify(), self.coeff * self.base.scalar)
         return Exp(self.base.simplify(), self.coeff)
+
+    def generator(self):
+        r"""Generator of an operator that is in single-parameter-form.
+
+        For example, for operator
+
+        .. math::
+
+            U(\phi) = e^{i\phi (0.5 Y + Z\otimes X)}
+
+        we get the generator
+
+        >>> U.generator()
+          (0.5) [Y0]
+        + (1.0) [Z0 X1]
+
+        """
+        if np.real(self.coeff) != 0 or not self.base.is_hermitian:
+            raise GeneratorUndefinedError(
+                f"Exponential with coefficient {self.coeff} and base operator {self.base} does not have a generator."
+            )
```

Comment:
I wonder if this condition is sufficient? Suppose `base = qml.s_prod(1j, qml.Identity(wires=0))` and `coeff=1`. In this case both conditions are True, but clearly the generator should be Identity for this exponential operator.

This might be resolved by using `qml.simplify`, but could add unwanted overhead.

Reply:
That is a good point, that would raise a GeneratorUndefinedError. If overhead is a concern, what if you don't simplify by default, but only to double-check? I.e.

```
  if np.real(self.coeff) == 0 and self.base.is_hermitian:
      return self.base

  simplified_op = qml.simplify(self)
  if np.real(simplified_op.coeff) == 0 and simplified_op.base.is_hermitian:
      return simplified_op.base
  
  raise GeneratorUndefinedError(
          f"Exponential with coefficient {self.coeff} and base ...
```

That way you only end up calling `qml.simplify` in instances where it's relevant, or where it's going to raise an error anyway.